### PR TITLE
Add --tarball to crane digest

### PR DESCRIPTION
**Pull request by jonjohnsonjr**
Fixes https://github.com/google/go-containerregistry/issues/895

### CODE REVIEW

**Comment by aelij** [comment]
Can you remove the required tag argument? Our tarballs have `"RepoTags": null` in their manifest.

**Comment by aelij** [comment]
@jonjohnsonjr Do you want me to finish this? I can create a sub-command as you suggested.

**Comment by aelij** [comment]
Ping :)

**Comment by jonjohnsonjr** [comment]
Ended up just finessing this with some manual flag logic. If we want to do something fancier in the future, we can revisit this.

I should probably provide a big warning about how this doesn't do what most people would expect, but we can do that if it becomes an issue.

File: cmd/crane/cmd/digest.go [hunk 1]
```diff
@@ -24,16 +24,42 @@ import (
...
+
+				img, err := crane.LoadTag(tarball, tag, *options...)
+				if err != nil {
+					log.Fatalf("loading image from %q: %v", tarball, err)
+				}
+				digest, err := img.Digest()
+				if err != nil {
+					log.Fatalf("computing digest: %v", err)
+				}
+				fmt.Println(digest.String())
```
Comment (ImJasonH) [inline cmd/crane/cmd/digest.go:47]: nit: return here, and unindent the `else` block. Maybe make both branches helper methods?

**Review by ImJasonH — APPROVED** [review]

File: cmd/crane/cmd/digest.go (hunk 1 above)
Comment (jonjohnsonjr) [reply cmd/crane/cmd/digest.go]: You drive a hard bargain.

**Review by ImJasonH — APPROVED** [review]

**Comment by aelij** [comment]
Thanks all

**Pull request merged**